### PR TITLE
bpo-43693: Compute deref offsets in compiler

### PR DESCRIPTION
**Pull request by markshannon**
Merges locals and cells into a single array.
Saves a pointer in the interpreter and means that we don't need the `LOAD_CLOSURE` opcode any more

https://bugs.python.org/issue43693

### CODE REVIEW

File: Python/ceval.c [hunk 1]
```diff
@@ -3047,7 +3046,7 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, PyFrameObject *f, int throwflag)
         }
 
         case TARGET(DELETE_DEREF): {
-            PyObject *cell = freevars[oparg];
+            PyObject *cell = GETLOCAL(oparg);
```
Comment (gvanrossum) [inline Python/ceval.c:3064]: Does this mean we can delete the GETLOCAL macro?

**Review by serhiy-storchaka — COMMENTED** [review]
Update also `Doc/library/dis.rst`.

**Comment by serhiy-storchaka** [comment]
Please don't forget to bump a bytecode magic number and add a line to the bytecode history list.

File: Python/ceval.c (hunk 1 above)
Comment (ericsnowcurrently) [reply Python/ceval.c:3064]: Probably, though this PR is a bit orthogonal to the question.

**Comment by gvanrossum** [comment]
Can we close this in favor of @ericsnowcurrently's #26216? Or are they complimentary?

**Comment by gvanrossum** [comment]
Never mind, #26216 is a cleanup PR that doesn't change any semantics.

**Comment by markshannon** [comment]
Eric is working on a PR to implement https://github.com/faster-cpython/ideas/issues/41, so I'm closing this.

**Comment by gvanrossum** [comment]
(What are you doing here?)

**Comment by ericsnowcurrently** [comment]
This change was always going to be included.  I figured it would still be cleaner if it were its own PR.  It also makes the subsequent PR a little smaller.  (Plus it was already reviewed, between Mark and me, so I'll merge it today. 🙂)

**Pull request merged**